### PR TITLE
added a source/network add/remove to/from zone for firewalld

### PR DESCRIPTION
**Pull request by t-v**
This adds the option to configure sources or networks in a zone.
I kept almost all the original coding and arguments, except for permanent.
Now it is no longer set as required, but it checks later on if you are using source or something else and defines the requirement there.

### CODE REVIEW

**Comment by ghost** [comment]
i think this is going to have to be rebased due to https://github.com/ansible/ansible-modules-extras/commit/62cae242cfc1948d60307cebb2c8473a4cd324d8 being merged.

**Comment by gregdek** [comment]
@maxamillion please review; if you +1, we will merge

**Comment by maxamillion** [comment]
I would like to see the docs updated to reflect the change in respect to the permanent argument but otherwise +1

**Comment by gregdek** [comment]
@maxamillion which docs need to be changed -- docs in the module or docs elsewhere?

**Comment by maxamillion** [comment]
@gregdek Docs in the module should be fine. As I understand it, that docstring is used to generate the module docs both in `ansible-doc` and at docs.ansible.com/modules

File: system/firewalld.py [hunk 1]
```diff
@@ -41,6 +41,11 @@
       - "Rich rule to add/remove to/from firewalld."
     required: false
     default: null
+  source:
```
Comment (bcoca) [inline system/firewalld.py:44]: needs version added (2.0)

**Comment by bcoca** [comment]
+1 to changes to module docs for 'permanent' field

**Comment by gregdek** [comment]
Sounds like the consensus is "make sure the documentation properly reflects the new behavior."  @TimothyVandenbrande if you can make these changes, we can have @maxamillion review again.

**Comment by t-v** [comment]
Due to an accident I'm unable to get to a computer, but once I can, I'll
update the docs.

**Comment by maxamillion** [comment]
@TimothyVandenbrande - I hope all is well, best wishes.

**Comment by t-v** [comment]
I added the version into the docs for the source variable and rebased my branch.

**Comment by t-v** [comment]
@maxamillion were there other things that you wished to see updated?

**Comment by maxamillion** [comment]
Looks good to me, +1 :+1:

**Comment by gregdek** [comment]
Merging. Thanks @TimothyVandenbrande for the PR and @maxamillion for the review!

**Pull request merged**